dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -65,13 +71,106 @@ public Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null
         {
             var role = splits[i];
             var content = splits[i + 1];
+            content = this.DecodeReservedCharIfNeeded(content);
             sb.Append("<message role=\"").Append(role).AppendLine("\">");
             sb.AppendLine(content);
             sb.AppendLine("</message>");
         }
 
         renderedResult = sb.ToString();
-
         return Task.FromResult(renderedResult);
     }
+
+    /// <summary>
+    /// pre-process the template before rendering.
+    /// If the template contains any reserved characters and <see cref="_allowUnsafeContent"/> is false,
+    /// throw an exception.
+    ///
+    /// Otherwise, no pre-processing is needed.
+    /// </summary>
+    /// <param name="template"></param>
+    /// <returns></returns>
+    private string PreProcessTemplate(string template)
+    {
+        if (this._allowUnsafeContent)
+        {
+            return template;
+        }
+
+        if (template.Contains(ReservedChar))
```

Comment:
What if my prompt template asks LLM to give me the history of the Ġ character?

Reply:
see https://github.com/microsoft/semantic-kernel/pull/6048#issuecomment-2085957297